lib/widget/movie_carousel.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+import 'package:the_movie_db/widget/movie_poster.dart';
+import '../utils/general_constants.dart';
+import '../views/movie_deck.dart';
+import '../models/movie.dart';
+import '../models/genre.dart';
+
+class MovieCarousel extends StatelessWidget {
+  final List<Movie> movies;
+  final List<Genre> genres;
+  final Genre genre;
+  static const double heightSizedBox = 350;
+
+  const MovieCarousel({
+    super.key,
+    required this.movies,
+    required this.genres,
+    required this.genre,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(
+            EdgeInsetsConst.edgeEight,
+          ),
+          child: Text(
+            genre.name,
+            style: const TextStyle(
+                fontSize: FontConst.fontTitle,
+                fontWeight: FontWeight.bold,
+                color: Colors.white),
+          ),
+        ),
+        SizedBox(
+          height: heightSizedBox,
+          child: ListView.builder(
+            scrollDirection: Axis.horizontal,
+            itemCount: movies.length,
+            itemBuilder: (context, index) {
+              final movie = movies[index];
+              return GestureDetector(
+                  onTap: () {
+                    Navigator.push(
+                      context,
+                      MaterialPageRoute(
+                        builder: (context) => MovieDeck(
+                          movie: movie,
+                          genres: genres,
```

Comment:
You're passing your genres across your entire widget tree. I wouldn't pass this list as a parameter, but get the genres again inside `MovieDeck` instead. You're even using caching in your repository!